From 35a94f99026f66d869efd922bb5d12da0445ffdf Mon Sep 17 00:00:00 2001
From: Sergio Aguirre <saaguirre@ti.com>
Date: Mon, 26 Jan 2009 15:19:46 -0600
Subject: [PATCH 223/364] OMAP3ISP: Backend: Always do workaround

This patch removes ISP_WORKAROUND checking, and also doesn't
Read anymore the memory read address, as this is going to
be configured in isp.c file.

Signed-off-by: Sakari Ailus <sakari.ailus@nokia.com>
Signed-off-by: Sergio Aguirre <saaguirre@ti.com>
---
 drivers/media/video/isp/ispresizer.c |   24 ++----------------------
 1 files changed, 2 insertions(+), 22 deletions(-)

diff --git a/drivers/media/video/isp/ispresizer.c b/drivers/media/video/isp/ispresizer.c
index 76035b1..d407bc6 100644
--- a/drivers/media/video/isp/ispresizer.c
+++ b/drivers/media/video/isp/ispresizer.c
@@ -25,10 +25,6 @@
 #include "ispreg.h"
 #include "ispresizer.h"
 
-#if ISP_WORKAROUND
-dma_addr_t buff_addr_lsc_wa;
-#endif
-
 /* Default configuration of resizer,filter coefficients,yenh for camera isp */
 static struct isprsz_yenh ispreszdefaultyenh = {0, 0, 0, 0};
 static struct isprsz_coef ispreszdefcoef = {
@@ -503,14 +499,8 @@ int ispresizer_config_size(u32 input_w, u32 input_h, u32 output_w,
 		return -EINVAL;
 	}
 
-#if ISP_WORKAROUND
-	buff_addr_lsc_wa = isp_buf_get();
-	if (buff_addr_lsc_wa) {
-		/* Set Resizer input address and offset adderss */
-		ispresizer_set_inaddr(buff_addr_lsc_wa);
-		ispresizer_config_inlineoffset(isp_reg_readl(OMAP3_ISP_IOMEM_PREV, ISPPRV_WADD_OFFSET));
-	}
-#endif
+	/* Set Resizer input address and offset adderss */
+	ispresizer_config_inlineoffset(isp_reg_readl(OMAP3_ISP_IOMEM_PREV, ISPPRV_WADD_OFFSET));
 
 	res = isp_reg_readl(OMAP3_ISP_IOMEM_RESZ, ISPRSZ_CNT) & (~(ISPRSZ_CNT_HSTPH_MASK |
 							ISPRSZ_CNT_VSTPH_MASK));
@@ -519,21 +509,11 @@ int ispresizer_config_size(u32 input_w, u32 input_h, u32 output_w,
 						ISPRSZ_CNT_VSTPH_SHIFT),
 						OMAP3_ISP_IOMEM_RESZ,
 						ISPRSZ_CNT);
-#if ISP_WORKAROUND
 	isp_reg_writel((0x00 << ISPRSZ_IN_START_HORZ_ST_SHIFT) |
 					(0x00 << ISPRSZ_IN_START_VERT_ST_SHIFT),
 					OMAP3_ISP_IOMEM_RESZ,
 					ISPRSZ_IN_START);
 
-#else
-	isp_reg_writel(((ispres_obj.ipwd_crop * 2) <<
-						ISPRSZ_IN_START_HORZ_ST_SHIFT) |
-						(ispres_obj.ipht_crop <<
-						ISPRSZ_IN_START_VERT_ST_SHIFT),
-						OMAP3_ISP_IOMEM_RESZ,
-						ISPRSZ_IN_START);
-#endif
-
 	isp_reg_writel((ispres_obj.inputwidth << ISPRSZ_IN_SIZE_HORZ_SHIFT) |
 						(ispres_obj.inputheight <<
 						ISPRSZ_IN_SIZE_VERT_SHIFT),
-- 
1.6.0.3

